### PR TITLE
ci: Check import formatting with `isort`

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/4241. WIP in PL as well in https://github.com/PyTorchLightning/pytorch-lightning/pull/4242.

### Description of the changes
- [x] Add `isort` config to `pyproject.toml` which is recommended in [docs](https://pycqa.github.io/isort/docs/configuration/config_files/)
- [x] Add CI check to `.github/workflows/code-format.yml`
- [x] Disable import check by other checkers if one exists
- [x] Apply `isort` to the codebase
- [x] Add `isort` pre-commit hook
- [x] ...?

Let me know if I should split this PR!

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by akihironitta** [comment]
I'll wait for https://github.com/PyTorchLightning/pytorch-lightning-bolts/pull/315 to get merged or closed to avoid conflicts in `tests/`.

**Comment by Borda** [comment]
@akihironitta how is it going here? seems to be empty...

**Comment by akihironitta** [comment]
Not a strong opinion, but I prefer to use `force_single_line = 'True'` option as I think it looks neat. Any thoughts?
```python
-from pl_bolts.datasets import DummyDataset, RandomDataset, RandomDictDataset, RandomDictStringDataset
+from pl_bolts.datasets import DummyDataset
+from pl_bolts.datasets import RandomDataset
+from pl_bolts.datasets import RandomDictDataset
+from pl_bolts.datasets import RandomDictStringDatase
```

**Comment by Borda** [comment]
> ```python
> from pl_bolts.datasets import DummyDataset, RandomDataset, RandomDictDataset, RandomDictStringDataset
> ```

if it loading from a single file, I would prefer a single line...

File: .github/workflows/code-format.yml [hunk 1]
```diff
@@ -15,14 +15,26 @@ jobs:
...
-          # python -m pip install --upgrade --user pip
           pip install flake8
           pip --version
         shell: bash
-
       - name: PEP8
         run: |
           flake8 .
+
+  isort:
```
Comment (Borda) [inline .github/workflows/code-format.yml:27]:
```suggestion
  imports-check-isort:
```

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@akihironitta I would welcome adding something similar also for PL, just in smaller steps, meaning one PR per PL/sub-package and then in the config enable one-by-one...

**Comment by akihironitta** [comment]
> @akihironitta I would welcome adding something similar also for PL, just in smaller steps, meaning one PR per PL/sub-package and then in the config enable one-by-one...

@Borda Sure!

**Pull request merged**